bridge/src/worker.rs

```diff
@@ -46,80 +38,54 @@ pub enum Response {
 pub type WorkerResult = Result<Vec<u8>, WorkerError>;
 type WorkerCallback = Box<dyn FnOnce(WorkerResult) + Send + 'static>;
 
-async fn handle_request(request: Request, outer_channel: SyncSender<Response>) {
-    match request {
-        Request::PollActivityTask { core_worker, callback } => {
-            let task = core_worker.poll_activity_task().await;
+async fn poll_activity_task_async(worker: Worker, callback: WorkerCallback) {
```

Comment:
I think you should just move the contents of this into `Worker::poll_activity_task_async` under spawn. If you really want to keep as a separate function, still could but make it in the `impl Worker`, no need to be top level. (but I think it's small enough that can just be inside that call)